env/fs_posix.cc

```diff
@@ -1043,6 +1043,87 @@ class PosixFileSystem : public FileSystem {
   }
 #endif  // ROCKSDB_IOURING_PRESENT
 
+  virtual IOStatus Poll(std::vector<void*>& io_handles,
+                        size_t min_completions) override {
+    IOStatus io_s;
+#if defined(ROCKSDB_IOURING_PRESENT)
+    // io_uring_queue_init.
+    struct io_uring* iu = nullptr;
+    if (thread_local_io_urings_) {
+      iu = static_cast<struct io_uring*>(thread_local_io_urings_->Get());
+      if (iu == nullptr) {
+        iu = CreateIOUring();
```

Comment:
This should not be necessary. No one should be calling Poll if they haven't already submitted a request successfully by calling ReadAsync. Its better to return NotSupported if iu is nullptr.